libpromises/var_expressions.c

```diff
@@ -95,6 +96,13 @@ static size_t VarRefHash(const VarRef *ref)
     return (h & (INT_MAX - 1));
 }
 
+unsigned int VarRefHash_untyped(const void *ref,
+                                unsigned int seed ARG_UNUSED,
+                                unsigned int max)
+{
+    return VarRefHash(ref) & (max - 1);
+}
```

Comment:
I might be completely misunderstanding what you're trying to do here, but shouldn't this be XOR (`^`)? Otherwise you risk a hash that is very biased towards `1` bits, since they are anded.